internal/services/hdinsight/hdinsight_cluster_pool_resource.go

```diff
@@ -0,0 +1,464 @@
+					},
+
+					"private_api_server_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+					},
+
+					"outbound_type": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(hdinsights.OutboundTypeLoadBalancer),
+							string(hdinsights.OutboundTypeUserDefinedRouting),
+						}, false),
+					},
+				},
+			},
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r ClusterPoolResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ClusterPoolResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ClusterPoolModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			client := metadata.Client.HDInsight2024.Hdinsights
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			id := hdinsights.NewClusterPoolID(subscriptionId, model.ResourceGroup, model.Name)
+
+			existing, err := client.ClusterPoolsGet(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			clusterPool := hdinsights.ClusterPool{
+				Location: location.Normalize(model.Location),
+				Properties: &hdinsights.ClusterPoolResourceProperties{
+					ClusterPoolProfile:       expandClusterPoolProfile(model.ClusterPoolProfile),
+					ComputeProfile:           expandComputeProfile(model.ComputeProfile),
+					LogAnalyticsProfile:      expandLogAnalyticsProfile(model.LogAnalyticsProfile),
+					ManagedResourceGroupName: pointer.To(model.ManagedResourceGroupName),
+					NetworkProfile:           expandNetworkProfile(model.NetworkProfile),
+				},
+				Tags: tags.Expand(model.Tags),
+			}
+
+			if err := client.ClusterPoolsCreateOrUpdateThenPoll(ctx, id, clusterPool); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ClusterPoolResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.HDInsight2024.Hdinsights
+
+			id, err := hdinsights.ParseClusterPoolID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state ClusterPoolModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding state: %+v", err)
+			}
+
+			existing, err := client.ClusterPoolsGet(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			model := existing.Model
```

Comment:
Shouldn't we check to see if the model is nil before we start accessing it's members?
```suggestion
			model := existing.Model
			if model == nil {
				return fmt.Errorf("%s model was `nil`", id)
			}
```